src/emmodel/model.py

```diff
@@ -126,4 +126,75 @@ def plot_time_trend(ax: plt.Axes, df: pd.DataFrame,
     ax.set_xlabel("time")
     for time in year_heads:
         ax.axvline(time, linestyle="--", color="gray")
-    return ax
\ No newline at end of file
+    return ax
+
+
+class LinearExcessMortalityModel:
+    """LinearExcessMortalityModel"""
+    def __init__(self, df: pd.DataFrame, col_obs: str, 
+                 stage1_col_covs: List[str], stage2_col_covs: List[str],
+                 col_covs_priors: List[str], model_type: str = 'two-stage'):
+        super(LinearExcessMortalityModel, self).__init__()
+        self.df = df
+        self.col_obs = col_obs
+        self.stage1_col_covs = stage1_col_covs
+        self.stage2_col_covs = stage2_col_covs
+        self.model_type = model_type
+        self.col_covs_priors = col_covs_priors
+
+    def run(self, prior_min_var: float = 0.1):
+        if self.model_type == 'two-stage':
+            self.run_stage_1()
+            self.priors = self.result_to_priors(min_var=prior_min_var)
+            self.run_stage_2()
+
+    def run_stage_1(self):
+        self.data = regmod.data.Data(col_obs=self.col_obs, col_covs=self.stage1_col_covs)
```

Comment:
it is better to create all the attributes in the `__init__` function :)

Reply:
yeah, i could do that. I was thinking there might be too many variables in the __init__, since there are two stages and most names look similar